utils/fields.js

```diff
@@ -1,23 +1,34 @@
 var fields = exports;
 
-fields.primitiveTypes = [
-  'char'
-, 'byte'
-, 'bool'
-, 'int8'
-, 'uint8'
-, 'int16'
-, 'uint16'
-, 'int32'
-, 'uint32'
-, 'int64'
-, 'uint64'
-, 'float32'
-, 'float64'
-, 'string'
-, 'time'
-, 'duration'
-];
+/* map of all primitive types and their default values */
+var map = {
+  'char': '',
+  'byte': 0,
+  'bool': false,
+  'int8': 0,
+  'uint8': 0,
+  'int16': 0,
+  'uint16': 0,
+  'int32': 0,
+  'uint32': 0,
+  'int64': 0,
+  'uint64': 0,
+  'float32': 0,
+  'float64': 0,
+  'string': '',
+  'time': 0,
+  'duration': 0
+};
+
+fields.primitiveTypes = Object.keys(map);
+
+fields.getDefaultValue = function(type) {
+  if (type.indexOf("[]") == type.length - 2) {
```

Comment:
typically if this is a constant length array (e.g. `float64[9] K` in `sensor_msgs/CameraInfo`) it would be set here as well.
```
getDefaultValue ('float64[9]')
  => return new Array(9).fill(getDefaultValue('float64');
```